src/metadataGeneration/parameterGenerator.ts

```diff
@@ -107,7 +107,7 @@ export class ParameterGenerator {
     };
   }
 
-  private getQueryParameter(parameter: ts.ParameterDeclaration): Tsoa.Parameter {
+  private getQueryParameter(parameter: ts.ParameterDeclaration): Tsoa.Parameter | null {
```

Comment:
Out of curiosity.. why did you go with `null` instead of `undefined`?

Reply:
I chose `null` because to me that means `there is no query parameter`, where `undefined` would mean something closer to `I'm not sure if there is a query parameter`. I also saw [this similar PR](https://github.com/lukeautry/tsoa/pull/310/files#diff-617900d54e0562ba65c1a694adb27dddR18) which used `null` too.

Happy to swap to undefined if that's preferred though!

Reply:
If it's already used in other, similar places, we can just leave it. Personally I prefer `undefined`, because to me it's more in line with optional parameters.